test/integration/dockerregistryclient_test.go

```diff
@@ -4,11 +4,68 @@ package integration
 
 import (
 	"reflect"
+	"strings"
 	"testing"
+	"time"
 
 	"github.com/openshift/origin/pkg/dockerregistry"
 )
 
+const (
+	pulpRegistryName        = "registry.access.redhat.com"
+	dockerHubV2RegistryName = "index.docker.io"
+	dockerHubV1RegistryName = "registry.hub.docker.com"
+	quayRegistryName        = "quay.io"
+
+	maxRetryCount = 4
+	retryAfter    = time.Millisecond * 500
+)
+
+// retryWhenUnreachable invokes given function several times until it succeeds,
+// returns network unrelated error or maximum number of attempts is reached.
+// Should be used to wrap calls to remote registry to prevent test failures
+// because of short-term outages.
+func retryWhenUnreachable(t *testing.T, failOnError bool, f func() error) error {
+	timeout := retryAfter
+	for attempt, err := 0, f(); err != nil; attempt, err = attempt+1, f() {
```

Comment:
I find this construction difficult to read, but I don't feel strongly enough in test code to force it to change.